alerter/ThresholdAlerterTest.java

```diff
@@ -1,27 +1,26 @@
-public class Alerter {
-    static int alertFailureCount = 0;
-    static int networkAlertStub(float celcius) {
-        System.out.println("ALERT: Temperature is " + celcius + " celcius");
-        // Return 200 for ok
-        // Return 500 for not-ok
-        // stub always succeeds and returns 200
-        return 200;
-    }
-    static void alertInCelcius(float farenheit) {
-        float celcius = (farenheit - 32) * 5 / 9;
-        int returnCode = networkAlertStub(celcius);
-        if (returnCode != 200) {
-            // non-ok response is not an error! Issues happen in life!
-            // let us keep a count of failures to report
-            // However, this code doesn't count failures!
-            // Add a test below to catch this bug. Alter the stub above, if needed.
-            alertFailureCount += 0;
-        }
-    }
-    public static void main(String[] args) {
-        alertInCelcius(400.5f);
-        alertInCelcius(303.6f);
-        System.out.printf("%d alerts failed.\n", alertFailureCount);
-        System.out.println("All is well (maybe!)\n");
-    }
-}
+
+
+public class ThresholdAlerterTest {
+
+    static int alertFailureCount = 0;
+
+    static void alertInCelcius(float farenheit,ThresholdAlerterStub thresholdAlerterStub) {
+        float celcius = (farenheit - 32) * 5 / 9;
+        int returnCode = thresholdAlerterStub.networkAlertStub(celcius);
+        if (returnCode != 200) {
+            // non-ok response is not an error! Issues happen in life!
+            // let us keep a count of failures to report
+            // However, this code doesn't count failures!
+            // Add a test below to catch this bug. Alter the stub above, if needed.
+            alertFailureCount += 0;
+            assert (alertFailureCount > 0):"Alert Failure Count Tracker Didn't Update";
+        }
+    }
+    public static void main(String[] args) {
+        ThresholdAlerterStub thresholdAlerterStub = new ThresholdAlerterStubImpl();
+        alertInCelcius(400.5f,thresholdAlerterStub);
+        alertInCelcius(303.6f,thresholdAlerterStub);
+        System.out.printf("%d alerts failed.\n", alertFailureCount);
```

Comment:
A check or assert on the value of alertfailureCount is needed to verify if alertInCelcius() method is working as expected.
for example, choose temperature values that are within and beyond threshold, with which we know what should be the value of alertFailureCount and then assert the required value.